[Synthetics] Private locations spaces constraint

cesco-f opened the pull request:
This PR is part of issue https://github.com/elastic/kibana/issues/208549.

Now that agent policies are space aware private locations spaces must be among the ones of the agent policy.

How to test it:

- Checkout main
- Create an agent policy in the default space
- Create a second space
- From the second space create a private location using the agent policy created in the default space
- Checkout this PR
- From the second space check that in the private location page the private location created in step 2 is still available and that the agent policy column says "Policy not found in the current space".
- Check that the agent policy in the default space is no longer available when creating a private location in the second space
- Create an agent policy in the second space
- Create a second private location using this policy, check that only the second space is available in the list of spaces of the private location
- Back to the default space, check that the policy created in the second space is not available when creating a private location

Before

https://github.com/user-attachments/assets/bdb8667a-bdb7-4b57-a655-7cdd9f98a965


After

https://github.com/user-attachments/assets/260cf01a-dcad-40e5-bae2-1316ff844486

cesco-f committed:
fix(private locations): private locations spaces must be among the one of the agent policy

cesco-f committed:
feat(api): handled api part

cesco-f committed:
test(fix): fix API test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335001)
* Commit: 5f03d60ac05ef82c84d62751411f63faf7c97a06

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5201eede172f77d9c66f0de2e00306667737aa20) | [after](https://github.com/elastic/kibana/commit/5f03d60ac05ef82c84d62751411f63faf7c97a06) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +168.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5201eede172f77d9c66f0de2e00306667737aa20) | [after](https://github.com/elastic/kibana/commit/5f03d60ac05ef82c84d62751411f63faf7c97a06) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.2KB | 28.3KB | +71.0B |



### History
* :broken_heart: [Build #334817](https://buildkite.com/elastic/kibana-pull-request/builds/334817) failed 1a8219f2acf0357d81cb272e98f3942598185044
* :broken_heart: [Build #334702](https://buildkite.com/elastic/kibana-pull-request/builds/334702) failed ec904f07c55e79307e2aa97e2a6b0944e16fe377

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335001","number":335001,"commit":"5f03d60ac05ef82c84d62751411f63faf7c97a06"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334817","number":334817,"commit":"1a8219f2acf0357d81cb272e98f3942598185044"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334702","number":334702,"commit":"ec904f07c55e79307e2aa97e2a6b0944e16fe377"}],"number":335001}
buildkite-pr-comment-->

shahzad31 commented on x-pack/solutions/observability/plugins/synthetics/server/routes/settings/private_locations/get_agent_policies.ts:
```diff
@@ -46,5 +46,6 @@ export const getAgentPoliciesAsInternalUser = async ({
     status: agentPolicy.status,
     description: agentPolicy.description,
     namespace: agentPolicy.namespace,
+    spaceIds: agentPolicy.space_ids || [agentPolicy.namespace],
```
agentPolicy.namespace isn't necessarily same as space id.

cesco-f replied:
What should be the value of `spaceIds` if `agentPolicy.space_ids` is `undefined`?

cesco-f committed:
feat(spaces): if no spaces defined for location policy spaces are used

cesco-f committed:
test(spaces): add test for undefined or empty spaces

cesco-f committed:
feat(all agent policies): agent policies from all spaces are retrieved

shahzad31 committed:
use proper space

shahzad31 committed:
use proper space

shahzad31 committed:
update bulk update

juliaElastic reviewed (approved):
Fleet changes LGTM

cesco-f committed:
Revert "update bulk update"

This reverts commit f203bb0b687cace7eb1a766b85284102ca1483dd.

cesco-f committed:
Revert "use proper space"

This reverts commit 6679097693f26d6912809410b8142133fb6dd8a2.

cesco-f committed:
Revert "use proper space"

This reverts commit 9b330bd08a71b8ca85ff6e17c55d45c2a1a072be.

cesco-f commented:
@elasticmachine merge upstream

elasticmachine commented:
merge conflict between base and head

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(create private location): fix create private location in non default space

cesco-f committed:
fix(location help text): slightly modified message

cesco-f committed:
feat(legacy agent policy): added endpoint to fetch legacy agent policy

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346401)
* Commit: 7ac1219087bcbe0fd1f90df7ef753c0c928b2915

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/346401#0199a08e-b6e2-49dc-aa0c-3fc1bf1c3840)

### History
* :yellow_heart: [Build #342591](https://buildkite.com/elastic/kibana-pull-request/builds/342591) was flaky 11d0ee2493e533868d90d475ffb6942ab11f522d
* :broken_heart: [Build #342503](https://buildkite.com/elastic/kibana-pull-request/builds/342503) failed bb2010cc078dd2be7222734232b0a5cf81455c6b
* :broken_heart: [Build #342417](https://buildkite.com/elastic/kibana-pull-request/builds/342417) failed 44691e2ea0cc98637742b112b222ddd80f7528f4
* :broken_heart: [Build #341791](https://buildkite.com/elastic/kibana-pull-request/builds/341791) failed 653763c7aa827eb563cd557fba89317bdf48622f
* :broken_heart: [Build #337755](https://buildkite.com/elastic/kibana-pull-request/builds/337755) failed f203bb0b687cace7eb1a766b85284102ca1483dd
* :green_heart: [Build #337036](https://buildkite.com/elastic/kibana-pull-request/builds/337036) succeeded e86de7a5b45af0412255d78abb3878cd1cf1c59c

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346401","number":346401,"commit":"7ac1219087bcbe0fd1f90df7ef753c0c928b2915"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342591","number":342591,"commit":"11d0ee2493e533868d90d475ffb6942ab11f522d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342503","number":342503,"commit":"bb2010cc078dd2be7222734232b0a5cf81455c6b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342417","number":342417,"commit":"44691e2ea0cc98637742b112b222ddd80f7528f4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341791","number":341791,"commit":"653763c7aa827eb563cd557fba89317bdf48622f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337755","number":337755,"commit":"f203bb0b687cace7eb1a766b85284102ca1483dd"}],"number":346401}
buildkite-pr-comment-->

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(test): fixed test

cesco-f committed:
fix(message): updated legacy policy message

cesco-f committed:
fix(revert): reverted change

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346565)
* Commit: c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2

### Failed CI Steps
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/346565#0199a41c-af05-4469-ad70-16e42b409261)
* [FTR Configs #2](https://buildkite.com/elastic/kibana-pull-request/builds/346565#0199a41c-af0f-4e2c-8237-18d43690e25d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/346565#0199a41c-af05-4469-ad70-16e42b409261) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/346565/jobs/0199a41c-af05-4469-ad70-16e42b409261/artifacts/0199a44f-6d4a-4aaf-aa7c-273453b730fe) Jest Tests #<span></span>10 / EQL Tab rendering pagination should load notes for current page only

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f8536178f3d8b89333fa978fcb7450a539f2cf14) | [after](https://github.com/elastic/kibana/commit/c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1369 | 1370 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f8536178f3d8b89333fa978fcb7450a539f2cf14) | [after](https://github.com/elastic/kibana/commit/c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +1.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f8536178f3d8b89333fa978fcb7450a539f2cf14) | [after](https://github.com/elastic/kibana/commit/c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.2KB | 28.3KB | +71.0B |



### History
* :broken_heart: [Build #346439](https://buildkite.com/elastic/kibana-pull-request/builds/346439) failed b8de4bc3f7afd9aa4e9810b704986dd383c1ca0d
* :broken_heart: [Build #346401](https://buildkite.com/elastic/kibana-pull-request/builds/346401) failed 7ac1219087bcbe0fd1f90df7ef753c0c928b2915
* :yellow_heart: [Build #342591](https://buildkite.com/elastic/kibana-pull-request/builds/342591) was flaky 11d0ee2493e533868d90d475ffb6942ab11f522d
* :broken_heart: [Build #342503](https://buildkite.com/elastic/kibana-pull-request/builds/342503) failed bb2010cc078dd2be7222734232b0a5cf81455c6b
* :broken_heart: [Build #342417](https://buildkite.com/elastic/kibana-pull-request/builds/342417) failed 44691e2ea0cc98637742b112b222ddd80f7528f4

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346565","number":346565,"commit":"c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346439","number":346439,"commit":"b8de4bc3f7afd9aa4e9810b704986dd383c1ca0d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346401","number":346401,"commit":"7ac1219087bcbe0fd1f90df7ef753c0c928b2915"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342591","number":342591,"commit":"11d0ee2493e533868d90d475ffb6942ab11f522d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342503","number":342503,"commit":"bb2010cc078dd2be7222734232b0a5cf81455c6b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342417","number":342417,"commit":"44691e2ea0cc98637742b112b222ddd80f7528f4"}],"number":346565}
buildkite-pr-comment-->

cesco-f committed:
fix(synthetics private location): synthetics private location is space aware

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346641)
* Commit: 0b943539078095405274498f7139769f2364f6d2

### Failed CI Steps
* [FTR Configs #119](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a49b-dac1-47e1-9aab-99caa833004f)
* [FTR Configs #119](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a4b7-e04a-4b3a-b9fe-3a7973972e80)
* [FTR Configs #121](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a49b-dac3-4dce-9183-173a0361e3a4)
* [FTR Configs #123](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a49b-dac5-408e-b20b-cd1c7c0ee418)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a49b-dac1-47e1-9aab-99caa833004f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/346641/jobs/0199a49b-dac1-47e1-9aab-99caa833004f/artifacts/0199a4b7-cb72-45e9-b9f5-6baadc7918e9) FTR Configs #<span></span>119 / Serverless Observability - Deployment-agnostic Synthetics API integration tests SyntheticsAPITests PrivateLocationCreateMonitor handles spaces
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a4b7-e04a-4b3a-b9fe-3a7973972e80) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/346641/jobs/0199a4b7-e04a-4b3a-b9fe-3a7973972e80/artifacts/0199a4c3-f1ec-41e3-b07c-77ec365f7368) FTR Configs #<span></span>119 / Serverless Observability - Deployment-agnostic Synthetics API integration tests SyntheticsAPITests PrivateLocationCreateMonitor handles spaces
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a49b-dac5-408e-b20b-cd1c7c0ee418) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/346641/jobs/0199a49b-dac5-408e-b20b-cd1c7c0ee418/artifacts/0199a4b7-ab9d-4bf4-b153-0d040adb0ebb) FTR Configs #<span></span>123 / Stateful Observability - Deployment-agnostic API integration tests SyntheticsAPITests PrivateLocationCreateMonitor handles spaces
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/346641#0199a49b-dac3-4dce-9183-173a0361e3a4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/346641/jobs/0199a49b-dac3-4dce-9183-173a0361e3a4/artifacts/0199a4c0-490a-4224-b6b8-737789ff94ad) FTR Configs #<span></span>121 / Synthetics API Tests PrivateLocationAddMonitor handles spaces

### History
* :yellow_heart: [Build #346565](https://buildkite.com/elastic/kibana-pull-request/builds/346565) was flaky c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2
* :broken_heart: [Build #346439](https://buildkite.com/elastic/kibana-pull-request/builds/346439) failed b8de4bc3f7afd9aa4e9810b704986dd383c1ca0d
* :broken_heart: [Build #346401](https://buildkite.com/elastic/kibana-pull-request/builds/346401) failed 7ac1219087bcbe0fd1f90df7ef753c0c928b2915
* :yellow_heart: [Build #342591](https://buildkite.com/elastic/kibana-pull-request/builds/342591) was flaky 11d0ee2493e533868d90d475ffb6942ab11f522d
* :broken_heart: [Build #342503](https://buildkite.com/elastic/kibana-pull-request/builds/342503) failed bb2010cc078dd2be7222734232b0a5cf81455c6b
* :broken_heart: [Build #342417](https://buildkite.com/elastic/kibana-pull-request/builds/342417) failed 44691e2ea0cc98637742b112b222ddd80f7528f4

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346641","number":346641,"commit":"0b943539078095405274498f7139769f2364f6d2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346565","number":346565,"commit":"c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346439","number":346439,"commit":"b8de4bc3f7afd9aa4e9810b704986dd383c1ca0d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346401","number":346401,"commit":"7ac1219087bcbe0fd1f90df7ef753c0c928b2915"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342591","number":342591,"commit":"11d0ee2493e533868d90d475ffb6942ab11f522d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342503","number":342503,"commit":"bb2010cc078dd2be7222734232b0a5cf81455c6b"}],"number":346641}
buildkite-pr-comment-->

shahzad31 commented on x-pack/solutions/observability/plugins/synthetics/server/synthetics_service/private_location/synthetics_private_location.ts:
```diff
@@ -362,7 +366,9 @@ export class SyntheticsPrivateLocation {
     spaceId: string
   ) {
-    const soClient = this.server.coreStart.savedObjects.createInternalRepository();
+    const soClient = this.server.coreStart.savedObjects
+      .getUnsafeInternalClient()
+      .asScopedToNamespace(spaceId);
```
Since before this PR package policy was always created in default space, this will not fetch existing package policies for a non default space. it will end up recreating those.

cesco-f committed:
test(fix): fixed tests

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

csr reviewed (approved):
`x-pack/solutions/observability/test/api_integration_deployment_agnostic/services/synthetics_private_location.ts` LGTM 👍

cesco-f committed:
refactor(package policy service): created PackagePolicyService

shahzad31 commented on x-pack/solutions/observability/plugins/synthetics/server/synthetics_service/private_location/package_policy_service.ts:
```diff
@@ -0,0 +1,131 @@
+export class PackagePolicyService {
+  private readonly server: SyntheticsServerSetup;
+
+  constructor(_server: SyntheticsServerSetup) {
```
```suggestion
  constructor(_server: Dependencies) {
```
i wish we can do this as well, hopefully in a follow up, we call it what it actually is.

cesco-f committed:
feat(migration): added migration of legacy agent policies

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347122)
* Commit: fcf520c661be50634ff3a09ed3fb5d9deee9d9f2

### Failed CI Steps
* [FTR Configs #136](https://buildkite.com/elastic/kibana-pull-request/builds/347122#0199aa36-e51c-4294-8fa0-91ef8d77974c)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/347122#0199aa36-e51c-4294-8fa0-91ef8d77974c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/347122/jobs/0199aa36-e51c-4294-8fa0-91ef8d77974c/artifacts/0199aa5c-90e6-4e85-ae07-531b81d1444a) FTR Configs #<span></span>136 / Serverless Common UI - Dev Tools Search Profiler Editor With a test index profiles a simple query

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/9756c91bc4992cf093504a870a39e7d925334703) | [after](https://github.com/elastic/kibana/commit/fcf520c661be50634ff3a09ed3fb5d9deee9d9f2) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1368 | 1369 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9756c91bc4992cf093504a870a39e7d925334703) | [after](https://github.com/elastic/kibana/commit/fcf520c661be50634ff3a09ed3fb5d9deee9d9f2) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +1.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9756c91bc4992cf093504a870a39e7d925334703) | [after](https://github.com/elastic/kibana/commit/fcf520c661be50634ff3a09ed3fb5d9deee9d9f2) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.2KB | 28.3KB | +71.0B |



### History
* :broken_heart: [Build #347066](https://buildkite.com/elastic/kibana-pull-request/builds/347066) failed 1e75163173e95685fd3eafb04c9bc34ea29c9c23
* :yellow_heart: [Build #346932](https://buildkite.com/elastic/kibana-pull-request/builds/346932) was flaky a68a691ad72a003f53abe52d85819058002905e6
* :broken_heart: [Build #346878](https://buildkite.com/elastic/kibana-pull-request/builds/346878) failed e990d1bf8f7e1c60fef741643529ffd21b35bd2a
* :broken_heart: [Build #346641](https://buildkite.com/elastic/kibana-pull-request/builds/346641) failed 0b943539078095405274498f7139769f2364f6d2
* :yellow_heart: [Build #346565](https://buildkite.com/elastic/kibana-pull-request/builds/346565) was flaky c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347122","number":347122,"commit":"fcf520c661be50634ff3a09ed3fb5d9deee9d9f2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347066","number":347066,"commit":"1e75163173e95685fd3eafb04c9bc34ea29c9c23"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346932","number":346932,"commit":"a68a691ad72a003f53abe52d85819058002905e6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346878","number":346878,"commit":"e990d1bf8f7e1c60fef741643529ffd21b35bd2a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346641","number":346641,"commit":"0b943539078095405274498f7139769f2364f6d2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346565","number":346565,"commit":"c4e83a1977b15136355a4a9cd61a0e0e82e7b1f2"}],"number":347122}
buildkite-pr-comment-->

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Revert "feat(legacy agent policy): added endpoint to fetch legacy agent policy"

This reverts commit 7ac1219087bcbe0fd1f90df7ef753c0c928b2915.

cesco-f committed:
Revert "feat(migration): added migration of legacy agent policies"

This reverts commit d19186794a92dcfd58c3f62ddb0df748ba4df444.

cesco-f committed:
fix(policy not found): changed message when policy not found

cesco-f committed:
fix(policies): policies must be found in current space

cesco-f committed:
fix(space id): spaceId is required in getAgentPoliciesAsInternalUser

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(legacy agent policies): package policy service always take the default space into consideration

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
feat(space awareness): agentPolicy.space_ids is not defined all spaces are considered

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(rebase): fixed rebase

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354559)
* Commit: 8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/76b55423c995d9c36c12fb230c65722dcbadafbf) | [after](https://github.com/elastic/kibana/commit/8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +216.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/76b55423c995d9c36c12fb230c65722dcbadafbf) | [after](https://github.com/elastic/kibana/commit/8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.2KB | 28.3KB | +71.0B |



### History
* :green_heart: [Build #351666](https://buildkite.com/elastic/kibana-pull-request/builds/351666) succeeded 14ab11caff14a97920c498e4e7cd6e197ec4a2e1
* :broken_heart: [Build #350028](https://buildkite.com/elastic/kibana-pull-request/builds/350028) failed b4481176d4656a95c091e1a8a70e981ea773d899
* :yellow_heart: [Build #348880](https://buildkite.com/elastic/kibana-pull-request/builds/348880) was flaky 1c898a911fab09ba11b0321ad3311592b03028c3
* :broken_heart: [Build #348628](https://buildkite.com/elastic/kibana-pull-request/builds/348628) failed 32c7364e5db4861c62ce4036364eabd30cbdd4b5
* :green_heart: [Build #348252](https://buildkite.com/elastic/kibana-pull-request/builds/348252) succeeded b050c47d673733c55736c2280b9cd83f55ed1b95
* :broken_heart: [Build #348202](https://buildkite.com/elastic/kibana-pull-request/builds/348202) failed 97b70a11b0b857a6ce3f360f4bad5968d3f27bd6

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354559","number":354559,"commit":"8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351666","number":351666,"commit":"14ab11caff14a97920c498e4e7cd6e197ec4a2e1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350028","number":350028,"commit":"b4481176d4656a95c091e1a8a70e981ea773d899"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348880","number":348880,"commit":"1c898a911fab09ba11b0321ad3311592b03028c3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348628","number":348628,"commit":"32c7364e5db4861c62ce4036364eabd30cbdd4b5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348252","number":348252,"commit":"b050c47d673733c55736c2280b9cd83f55ed1b95"}],"number":354559}
buildkite-pr-comment-->

justinkambic reviewed (commented):
I'm attempting to do the practical testing steps in the description, but I seem to have encountered an issue.

```
Checkout main
Create an agent policy in the default space
Create a second space
From the second space create a private location using the agent policy created in the default space
Checkout this PR
From the second space check that in the private location page the agent policy created in step 2 is still available <-- this does not work for me, I can't see the private location from the previous step(s) anymore
Check that the agent policy in the default space is no longer available when creating a private location in the second space
Create an agent policy in the second space
Create a second private location using this policy, check that only the second space is available in the list of spaces of the private location
Back to the default space, check that the policy created in the second space is not available when creating a private location
```

As I understand it, we're meant to still see the private location we defined in space 2, using an agent policy defined in the default space, on the main branch. Is that correct?

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(private locations): private locations are shown even if agent policy not found in the space

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

cesco-f committed:
fix(spaces select): if spaces select is disabled don't set spaces list

justinkambic commented:
@cesco-f this is looking pretty good. One question. When I create an Agent policy and specify `*` as the choice for spaces in advanced settings, if I then try to use that agent policy and reassign specific spaces on my location, this is not allowed. Is this intentional?

I made a policy that explicitly defines all the spaces and I can pick the space(s) to apply to the subsequent location. See screenshots:

## Using a policy with *

<img width="534" height="672" alt="image" src="https://github.com/user-attachments/assets/0c229654-f5d2-4de9-ae99-ebe5dd557f35" />


## Using a policy with all spaces explicitly set up

<img width="478" height="718" alt="image" src="https://github.com/user-attachments/assets/487b83ac-8ea3-4cdf-92f6-4ec519202d93" />

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
feat(all spaces): handle all spaces agent policy

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(rebase): fix rebase

justinkambic reviewed (approved):
I don't see any real code issues to hold this up. Practical testing has worked as intended. LGTM

cesco-f commented:
/ci

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357398)
* Commit: d2c42e5e78d350897e11a097992092995ff82283

### Failed CI Steps
* [FTR Configs #11](https://buildkite.com/elastic/kibana-pull-request/builds/357398#019a3ebf-74d0-48ca-94b8-ac406dc7df19)
* [FTR Configs #54](https://buildkite.com/elastic/kibana-pull-request/builds/357398#019a3ebf-74ef-45a7-baa0-504b5f21bc84)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/00fdd9178c80714d49137a8bc2fc09cc7f0c8844) | [after](https://github.com/elastic/kibana/commit/d2c42e5e78d350897e11a097992092995ff82283) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +279.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/00fdd9178c80714d49137a8bc2fc09cc7f0c8844) | [after](https://github.com/elastic/kibana/commit/d2c42e5e78d350897e11a097992092995ff82283) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.6KB | 28.6KB | +71.0B |



### History
* :green_heart: [Build #357070](https://buildkite.com/elastic/kibana-pull-request/builds/357070) succeeded 45d31c5ee321cd1267d8b20dfe9b86d4f0eaecbd
* :yellow_heart: [Build #355602](https://buildkite.com/elastic/kibana-pull-request/builds/355602) was flaky 6ab919e1322fbaf9b4c6a38bd385e9a27b47c5a0
* :green_heart: [Build #354559](https://buildkite.com/elastic/kibana-pull-request/builds/354559) succeeded 8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8
* :green_heart: [Build #351666](https://buildkite.com/elastic/kibana-pull-request/builds/351666) succeeded 14ab11caff14a97920c498e4e7cd6e197ec4a2e1
* :broken_heart: [Build #350028](https://buildkite.com/elastic/kibana-pull-request/builds/350028) failed b4481176d4656a95c091e1a8a70e981ea773d899
* :yellow_heart: [Build #348880](https://buildkite.com/elastic/kibana-pull-request/builds/348880) was flaky 1c898a911fab09ba11b0321ad3311592b03028c3

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357398","number":357398,"commit":"d2c42e5e78d350897e11a097992092995ff82283"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357070","number":357070,"commit":"45d31c5ee321cd1267d8b20dfe9b86d4f0eaecbd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355602","number":355602,"commit":"6ab919e1322fbaf9b4c6a38bd385e9a27b47c5a0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354559","number":354559,"commit":"8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351666","number":351666,"commit":"14ab11caff14a97920c498e4e7cd6e197ec4a2e1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350028","number":350028,"commit":"b4481176d4656a95c091e1a8a70e981ea773d899"}],"number":357398}
buildkite-pr-comment-->

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'main' into fix-private-locations-spaces

shahzad31 committed:
Pr feedback

shahzad31 committed:
Merge branch 'fix-private-locations-spaces' of https://github.com/cesco-f/kibana into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/359670)
* Commit: 22723f80211a53b722b2d7131dc4afdb7317c940

### Failed CI Steps
* [Jest Integration Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/359670#019a6cfb-951d-4b35-a3f1-b18fe801ab4c)
* [Jest Integration Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/359670#019a6d00-48d8-4c50-a6a6-9099fbb8d5b9)

### History
* :yellow_heart: [Build #359307](https://buildkite.com/elastic/kibana-pull-request/builds/359307) was flaky 185ffd282b4cd8e4deb14a125476ee2c5a30805c
* :yellow_heart: [Build #358071](https://buildkite.com/elastic/kibana-pull-request/builds/358071) was flaky 55429fb7300844f3700ed6345c03b5c5646e04ac
* :yellow_heart: [Build #357398](https://buildkite.com/elastic/kibana-pull-request/builds/357398) was flaky d2c42e5e78d350897e11a097992092995ff82283
* :green_heart: [Build #357070](https://buildkite.com/elastic/kibana-pull-request/builds/357070) succeeded 45d31c5ee321cd1267d8b20dfe9b86d4f0eaecbd
* :yellow_heart: [Build #355602](https://buildkite.com/elastic/kibana-pull-request/builds/355602) was flaky 6ab919e1322fbaf9b4c6a38bd385e9a27b47c5a0
* :green_heart: [Build #354559](https://buildkite.com/elastic/kibana-pull-request/builds/354559) succeeded 8a3ee380fc60b0eb0e7c7e14e02d493546d5fdc8

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359670","number":359670,"commit":"22723f80211a53b722b2d7131dc4afdb7317c940"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359307","number":359307,"commit":"185ffd282b4cd8e4deb14a125476ee2c5a30805c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358071","number":358071,"commit":"55429fb7300844f3700ed6345c03b5c5646e04ac"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357398","number":357398,"commit":"d2c42e5e78d350897e11a097992092995ff82283"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357070","number":357070,"commit":"45d31c5ee321cd1267d8b20dfe9b86d4f0eaecbd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355602","number":355602,"commit":"6ab919e1322fbaf9b4c6a38bd385e9a27b47c5a0"}],"number":359670}
buildkite-pr-comment-->

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
fix(get policies): in default space agent policies are not fetched

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/360298)
* Commit: b39cd78a3be936b6e856467ac9493bdd9e3f1d89

### Failed CI Steps
* [FTR Configs #32](https://buildkite.com/elastic/kibana-pull-request/builds/360298#019a7422-a9cb-46a7-841e-141dbadaf80b)
* [FTR Configs #108](https://buildkite.com/elastic/kibana-pull-request/builds/360298#019a7422-aa3d-4ef5-8df1-05fb5a700eb9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/360298#019a7422-a9cb-46a7-841e-141dbadaf80b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/360298/jobs/019a7422-a9cb-46a7-841e-141dbadaf80b/artifacts/019a744d-a741-4faf-9e62-589531999239) FTR Configs #<span></span>32 / Alerting builtin alertTypes es_query rule runs correctly: threshold on ungrouped hit count < >
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/360298#019a7422-aa3d-4ef5-8df1-05fb5a700eb9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/360298/jobs/019a7422-aa3d-4ef5-8df1-05fb5a700eb9/artifacts/019a744a-137b-46fb-a562-22790173a4d5) FTR Configs #<span></span>108 / serverless observability UI - Cases and Rules Serverless Observability Cases Cases list bulk actions status change the status of cases to in-progress correctly

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9b2cff682df9d48b80a7009a14166a75e2aa1526) | [after](https://github.com/elastic/kibana/commit/b39cd78a3be936b6e856467ac9493bdd9e3f1d89) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +551.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9b2cff682df9d48b80a7009a14166a75e2aa1526) | [after](https://github.com/elastic/kibana/commit/b39cd78a3be936b6e856467ac9493bdd9e3f1d89) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.6KB | 28.6KB | +71.0B |



### History
* :green_heart: [Build #359739](https://buildkite.com/elastic/kibana-pull-request/builds/359739) succeeded b37f961048205a57aeebbd5d1e3f3813de4d230e
* :broken_heart: [Build #359670](https://buildkite.com/elastic/kibana-pull-request/builds/359670) failed 22723f80211a53b722b2d7131dc4afdb7317c940
* :yellow_heart: [Build #359307](https://buildkite.com/elastic/kibana-pull-request/builds/359307) was flaky 185ffd282b4cd8e4deb14a125476ee2c5a30805c
* :yellow_heart: [Build #358071](https://buildkite.com/elastic/kibana-pull-request/builds/358071) was flaky 55429fb7300844f3700ed6345c03b5c5646e04ac
* :yellow_heart: [Build #357398](https://buildkite.com/elastic/kibana-pull-request/builds/357398) was flaky d2c42e5e78d350897e11a097992092995ff82283
* :green_heart: [Build #357070](https://buildkite.com/elastic/kibana-pull-request/builds/357070) succeeded 45d31c5ee321cd1267d8b20dfe9b86d4f0eaecbd

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360298","number":360298,"commit":"b39cd78a3be936b6e856467ac9493bdd9e3f1d89"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359739","number":359739,"commit":"b37f961048205a57aeebbd5d1e3f3813de4d230e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359670","number":359670,"commit":"22723f80211a53b722b2d7131dc4afdb7317c940"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359307","number":359307,"commit":"185ffd282b4cd8e4deb14a125476ee2c5a30805c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358071","number":358071,"commit":"55429fb7300844f3700ed6345c03b5c5646e04ac"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357398","number":357398,"commit":"d2c42e5e78d350897e11a097992092995ff82283"}],"number":360298}
buildkite-pr-comment-->

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

jloleysens reviewed (approved):
🌔

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368822)
* Commit: 19566e052fb8a00a410fb61eb227c50a26b62c55

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c8d2dcb214a40d4b7f6dc49b8cc4b67f6c17ee87) | [after](https://github.com/elastic/kibana/commit/19566e052fb8a00a410fb61eb227c50a26b62c55) | diff |
| --- | --- | --- | --- |
| `synthetics` | 1.0MB | 1.0MB | +551.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c8d2dcb214a40d4b7f6dc49b8cc4b67f6c17ee87) | [after](https://github.com/elastic/kibana/commit/19566e052fb8a00a410fb61eb227c50a26b62c55) | diff |
| --- | --- | --- | --- |
| `synthetics` | 28.8KB | 28.9KB | +71.0B |



### History
* :green_heart: [Build #368402](https://buildkite.com/elastic/kibana-pull-request/builds/368402) succeeded 02e30c25cd14b703dafebb4b1159a98c8ac04904
* :green_heart: [Build #367016](https://buildkite.com/elastic/kibana-pull-request/builds/367016) succeeded a4021dabeea6cd73694838f9cd9664eb7e2df444
* :yellow_heart: [Build #364845](https://buildkite.com/elastic/kibana-pull-request/builds/364845) was flaky c3f38464e46fd0c038942bdc0a292e00b4512327
* :broken_heart: [Build #360514](https://buildkite.com/elastic/kibana-pull-request/builds/360514) failed bfeec5a32d2970fbc49c63d4a1ed7ac3420b9f7c
* :yellow_heart: [Build #360298](https://buildkite.com/elastic/kibana-pull-request/builds/360298) was flaky b39cd78a3be936b6e856467ac9493bdd9e3f1d89
* :green_heart: [Build #359739](https://buildkite.com/elastic/kibana-pull-request/builds/359739) succeeded b37f961048205a57aeebbd5d1e3f3813de4d230e

cc @cesco-f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368822","number":368822,"commit":"19566e052fb8a00a410fb61eb227c50a26b62c55"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368402","number":368402,"commit":"02e30c25cd14b703dafebb4b1159a98c8ac04904"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367016","number":367016,"commit":"a4021dabeea6cd73694838f9cd9664eb7e2df444"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/364845","number":364845,"commit":"c3f38464e46fd0c038942bdc0a292e00b4512327"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360514","number":360514,"commit":"bfeec5a32d2970fbc49c63d4a1ed7ac3420b9f7c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360298","number":360298,"commit":"b39cd78a3be936b6e856467ac9493bdd9e3f1d89"}],"number":368822}
buildkite-pr-comment-->

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

cesco-f committed:
Merge branch 'main' into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'main' into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'main' into fix-private-locations-spaces

fkanout reviewed (approved):
Tested locally and it works as described 
<img width="852" height="411" alt="Screenshot 2026-01-08 at 13 17 06" src="https://github.com/user-attachments/assets/fe9fe3e2-6bc3-4f80-abd8-9ad966429f6d" />
<img width="1451" height="459" alt="Screenshot 2026-01-08 at 13 19 38" src="https://github.com/user-attachments/assets/2dc14bba-5490-424b-9391-8aa96d924dfb" />
<img width="1129" height="387" alt="Screenshot 2026-01-08 at 13 20 03" src="https://github.com/user-attachments/assets/00383ce8-2aef-42c4-ab52-d9acbb449f97" />

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

shahzad31 committed:
resolve conflicts

shahzad31 committed:
resolve conflicts

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

shahzad31 committed:
conflicts

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'fix-private-locations-spaces' of https://github.com/cesco-f/kibana into fix-private-locations-spaces

shahzad31 committed:
conflicts

shahzad31 committed:
Merge branch 'main' into fix-private-locations-spaces

kibanamachine committed:
Changes from node scripts/lint_packages --fix

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

shahzad31 committed:
types

shahzad31 committed:
Merge branch 'fix-private-locations-spaces' of https://github.com/cesco-f/kibana into fix-private-locations-spaces

shahzad31 committed:
remove extra files

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

shahzad31 committed:
update test

shahzad31 committed:
Merge branch 'fix-private-locations-spaces' of https://github.com/cesco-f/kibana into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

shahzad31 committed:
fix tests

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

shahzad31 committed:
conflicts

shahzad31 committed:
update unit tests

shahzad31 committed:
update scout tests

shahzad31 committed:
Merge branch 'main' of https://github.com/elastic/kibana into fix-private-locations-spaces

shahzad31 committed:
Merge branch 'main' into fix-private-locations-spaces

shahzad31 committed:
Enhance PackagePolicyService and PrivateLocationSchema

- Updated PackagePolicyService to streamline retrieval of package policies by consolidating space client logic.
- Modified the handling of agent policies to include those associated with ALL_SPACES_ID.
- Adjusted PrivateLocationSchema to enforce a maximum size of 100 for the spaces array.

These changes improve the efficiency and robustness of package policy management and validation in the synthetics service.
